### PR TITLE
Tools: Add store check sub command to list the stores with specified state

### PR DESCRIPTION
**Pull request by xuhui-lu**
Signed-off-by: xuhui-lu <luxuhui12345@126.com>

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — COMMENTED** [review]
Good job. And I think we also need a unit test to check it.

**Comment by nolouch** [comment]
Hi @xuhui-lu , you can add the test in `tests/pdctl/store/store_test.go`.

**Comment by xuhui-lu** [comment]
> Hi @xuhui-lu , you can add the test in `tests/pdctl/store/store_test.go`.

sure, will add that.

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 1]
```diff
@@ -101,6 +102,16 @@ func NewStoreLimitCommand() *cobra.Command {
 	return c
 }
 
+// NewStoreCheckCommand return a check subcommand of storeCmd
+func NewStoreCheckCommand() *cobra.Command {
+	d := &cobra.Command{
+		Use:   "check <store_status>",
```
Comment (nolouch) [inline tools/pd-ctl/pdctl/command/store_command.go:108]:
```suggestion
		Use:   "check [up|offline|Tombstone]",
```

Comment (xuhui-lu) [reply tools/pd-ctl/pdctl/command/store_command.go]: Thx! nice catch.

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 2]
```diff
@@ -447,6 +458,21 @@ func storeLimitCommandFunc(cmd *cobra.Command, args []string) {
 	}
 }
 
+func storeCheckCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		cmd.Usage()
+		return
+	}
+
+	prefix := storesPrefix + "?state=" + args[0]
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:467]: The state can not directly use "Tombstone"/"up"/"offline". See https://github.com/tikv/pd/blob/69209e0a3ec20db90242a9375cf885724440c68e/server/api/store.go#L655. ^ ^
I think we need to check the output result in your tests.

Comment (xuhui-lu) [reply tools/pd-ctl/pdctl/command/store_command.go]: Right, thx for your reminder. There is a mapping from string value to numeric value.

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 3]
```diff
@@ -101,6 +102,16 @@ func NewStoreLimitCommand() *cobra.Command {
 	return c
 }
 
+// NewStoreCheckCommand return a check subcommand of storeCmd
+func NewStoreCheckCommand() *cobra.Command {
+	d := &cobra.Command{
+		Use:   "check [up|offline|Tombstone]",
```
Comment (nolouch) [inline tools/pd-ctl/pdctl/command/store_command.go:108]:
```suggestion
		Use:   "check [up|offline|tombstone]",
```

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 4]
```diff
@@ -447,6 +458,28 @@ func storeLimitCommandFunc(cmd *cobra.Command, args []string) {
...
 }
 
+func storeCheckCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		cmd.Usage()
+		return
+	}
+
+	state := args[0]
+	stateValue, ok := metapb.StoreState_value[state]
```
Comment (nolouch) [inline tools/pd-ctl/pdctl/command/store_command.go:468]: I think we should make case insensitive

File: tests/pdctl/store/store_test.go [hunk 5]
```diff
@@ -327,4 +327,21 @@ func (s *storeTestSuite) TestStore(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(scene.Idle, Equals, 100)
 
+	// store check status
+	args = []string{"-u", pdAddr, "store", "check", "Offline"}
+	output, err = pdctl.ExecuteCommand(cmd, args...)
+	c.Assert(err, IsNil)
+	c.Assert(strings.Contains(string(output), "PANIC"), IsFalse)
```
Comment (rleungx) [inline tests/pdctl/store/store_test.go:334]: It's better to check whether the output contains the offline stores instead of `PANIC`.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm, thanks!

**Pull request merged**